event-handlers/src/main/java/no/sikt/nva/nvi/events/UpsertNviCandidateHandler.java

```diff
@@ -51,15 +53,19 @@ public Void handleRequest(SQSEvent input, Context context) {
 
     private static void validateMessage(CandidateEvaluatedMessage message) {
         attempt(() -> {
-            Objects.requireNonNull(message.publicationBucketUri());
             Objects.requireNonNull(message.candidate().publicationId());
             return message;
         }).orElseThrow(failure -> new InvalidNviMessageException(INVALID_NVI_CANDIDATE_MESSAGE));
     }
 
     private void upsertNviCandidate(CandidateEvaluatedMessage evaluatedCandidate) {
         validateMessage(evaluatedCandidate);
-        CandidateBO.fromRequest(evaluatedCandidate, repository, periodRepository);
+        if (evaluatedCandidate.candidate() instanceof NviCandidate candidate) {
+            CandidateBO.fromRequest(candidate, repository, periodRepository);
+        }
+        if (evaluatedCandidate.candidate() instanceof NonNviCandidate nonNviCandidate) {
+            CandidateBO.fromRequest(nonNviCandidate, repository);
+        }
```

Comment:
Weird — in my mind it either _is_ or _is not_ a candidate, here we potentially have another state _I neither confirm nor deny_.